From 1b5b2db0663167e2e80c6e74bd1e09b1b29efd41 Mon Sep 17 00:00:00 2001
From: John Ogness <john.ogness@linutronix.de>
Date: Mon, 19 Oct 2020 22:11:31 +0206
Subject: [PATCH 034/194] printk: combine boot_delay_msec() into printk_delay()

boot_delay_msec() is always called immediately before printk_delay()
so just combine the two.

Signed-off-by: John Ogness <john.ogness@linutronix.de>
Signed-off-by: Sebastian Andrzej Siewior <bigeasy@linutronix.de>
---
 kernel/printk/printk.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/kernel/printk/printk.c b/kernel/printk/printk.c
index 758fffab1ba9..d79f5e6ec0eb 100644
--- a/kernel/printk/printk.c
+++ b/kernel/printk/printk.c
@@ -1753,8 +1753,10 @@ static void put_sprint_buf(int id, unsigned long flags)
 
 int printk_delay_msec __read_mostly;
 
-static inline void printk_delay(void)
+static inline void printk_delay(int level)
 {
+	boot_delay_msec(level);
+
 	if (unlikely(printk_delay_msec)) {
 		int m = printk_delay_msec;
 
@@ -2090,8 +2092,7 @@ asmlinkage int vprintk_emit(int facility, int level,
 		in_sched = true;
 	}
 
-	boot_delay_msec(level);
-	printk_delay();
+	printk_delay(level);
 
 	/* This stops the holder of console_sem just where we want him */
 	printed_len = vprintk_store(facility, level, dev_info, fmt, args);
-- 
2.19.1

